File: app/src/main/java/com/example/projetoestagioaiko/api/RetrofitClient.java
package com.example.projetoestagioaiko.api;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    private static Retrofit retrofit = null;
    private static final String BASE_URL = "https://aiko-olhovivo-proxy.aikodigital.io/";
    public static Retrofit getClient(String baseUrl) {
            if(retrofit == null){
               retrofit = new Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build();

            }
            return retrofit;
    }

}

File: app/src/main/java/br/com/okayamafilho/testesptrans/data/RetrofitClient.kt
package br.com.okayamafilho.testesptrans.data

import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {
    val spTransAPI by lazy {
        Retrofit.Builder()
            .baseUrl("http://api.olhovivo.sptrans.com.br/v2.1/")
            .addConverterFactory(GsonConverterFactory.create())
            .client(
                OkHttpClient.Builder().build()
            )
            .build()
            .create(SPTransAPI::class.java)
    }
}